### PR TITLE
Task/119 - Ekran Kariera

### PR DESCRIPTION
**Pull request by Kuba-Nalepa**

### CODE REVIEW

File: app/src/main/java/com/safetyheads/akademiaandroida/career/CareerFragment.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        binding.notificationView.switchButtonListener {
+            viewModel.writeSetting(Settings.SEND_NOTIFICATIONS, it)
+        }
+    }
+
+    override fun onButtonClick(jobUrl: String, position: Int) {
+        val myWebView = binding.webView
+        myWebView.loadUrl(jobUrl)
+        binding.notificationView.visibility = View.GONE
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/career/CareerFragment.kt:66]: where do you show those again when getting back from the job offer?

File: app/src/main/java/com/safetyheads/akademiaandroida/views/notification_view/NotificationsView.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                binding.expandableContent.visibility = View.GONE
+            }
+        }
+    }
+
+    fun setText(resIdText: Int) {
+        binding.titleText.text = resources.getString(resIdText)
+    }
+
+    fun switchButtonListener(x: (isChecked: Boolean) -> Any) {
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/views/notification_view/NotificationsView.kt:44]: x is not a good name, please name in meaningfully

**Review by przemek19980102 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/safetyheads/akademiaandroida/career/CareerFragment.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentCareerBinding.inflate(layoutInflater)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.notificationView.initOnClickListeners()
+        binding.recyclerView.layoutManager = LinearLayoutManager(requireContext())
```
Comment (rafalkrystek) [inline app/src/main/java/com/safetyheads/akademiaandroida/career/CareerFragment.kt:31]: can be added through xml

File: app/src/main/java/com/safetyheads/akademiaandroida/settings/SettingRepositoryImpl.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package com.safetyheads.akademiaandroida.settings
+
+import android.content.Context
+import com.safetyheads.domain.entities.Settings
+import com.safetyheads.domain.repositories.SettingsRepository
+
+class SettingRepositoryImpl(val context: Context): SettingsRepository {
+
+    private val sharedPreferences = context.getSharedPreferences("Settings", Context.MODE_PRIVATE)
+    private val editor = sharedPreferences.edit()
```
Comment (rafalkrystek) [inline app/src/main/java/com/safetyheads/akademiaandroida/settings/SettingRepositoryImpl.kt:10]: unused

File: app/src/main/java/com/safetyheads/akademiaandroida/views/notification_view/NotificationsView.kt [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+): FrameLayout(context, attrs, defStyle) {
+
+    private val binding = ViewNotificationsBinding.inflate(
+        LayoutInflater.from(context),
+        this,
+        true
+    )
+
+
+    fun initOnClickListeners() {
```
Comment (rafalkrystek) [inline app/src/main/java/com/safetyheads/akademiaandroida/views/notification_view/NotificationsView.kt:25]: Maybe init block could work better? This way you woldnt need to call this method from CareerFragment

File: app/src/main/res/layout/view_notifications.xml [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    style="@style/default_card_style">
+
+    <LinearLayout
+        android:id="@+id/linearLayout"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="vertical"
+        android:animateLayoutChanges="true">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (rafalkrystek) [inline app/src/main/res/layout/view_notifications.xml:14]: Multiple constraint layouts in single xml mixed with LinearLayouts seems like an overkill.
ConstraintLayout should help with keeping layout "flat"

File: app/src/main/res/values/attrs_notifications_view.xml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<resources>
+    <declare-styleable name="NotificationsView">
```
Comment (rafalkrystek) [inline app/src/main/res/values/attrs_notifications_view.xml:2]: I didnt see use for that customs atributes

File: domain/src/main/java/com/safetyheads/domain/repositories/CareerRepository.kt [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package com.safetyheads.domain.repositories
+
+import com.safetyheads.domain.entities.JobOffer
+import kotlinx.coroutines.flow.Flow
+
+interface CareerRepository {
+
+    suspend fun getJobOffersList(): Flow<ArrayList<JobOffer>>
```
Comment (msarelo) [inline domain/src/main/java/com/safetyheads/domain/repositories/CareerRepository.kt:8]:
```suggestion
    suspend fun getJobOffersList(): Flow<List<JobOffer>>
```

File: app/src/main/java/com/safetyheads/akademiaandroida/career/CareerFragment.kt (hunk 1 above)
Comment (Kuba-Nalepa) [reply app/src/main/java/com/safetyheads/akademiaandroida/career/CareerFragment.kt]: created new activity, which can easy navigate back, by back button

**Pull request merged**